File: VB/DeterminePrinterSettings/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports System.Drawing.Printing
Imports DevExpress.XtraPrinting
' ...

Namespace DeterminePrinterSettings
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim report As New XtraReport1()
			AddHandler report.PrintingSystem.StartPrint, AddressOf PrintingSystem_StartPrint
			report.ShowPreviewDialog()
			RemoveHandler report.PrintingSystem.StartPrint, AddressOf PrintingSystem_StartPrint
		End Sub

		Private Sub PrintingSystem_StartPrint(ByVal sender As Object, ByVal e As PrintDocumentEventArgs)
			AddHandler e.PrintDocument.BeginPrint, AddressOf PrintDocument_BeginPrint
		End Sub

		Private Sub PrintDocument_BeginPrint(ByVal sender As Object, ByVal e As PrintEventArgs)
			Dim doc As PrintDocument = TryCast(sender, PrintDocument)
			Dim printerName As String = doc.PrinterSettings.PrinterName
			' ... your code
		End Sub

	End Class
End Namespace

File: CS/DeterminePrinterSettings/Form1.cs
using System;
using System.Drawing.Printing;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using DevExpress.XtraReports.UI;
// ...

namespace DeterminePrinterSettings {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();
            ReportPrintTool printTool = new ReportPrintTool(report);

            ((XtraReport1)printTool.Report).PrintingSystem.StartPrint +=
                new PrintDocumentEventHandler(PrintingSystem_StartPrint);

            printTool.ShowPreviewDialog();

            ((XtraReport1)printTool.Report).PrintingSystem.StartPrint -= PrintingSystem_StartPrint;
        }

        private void PrintingSystem_StartPrint(object sender, PrintDocumentEventArgs e) {
            e.PrintDocument.BeginPrint +=
                new PrintEventHandler(PrintDocument_BeginPrint);
        }

        private void PrintDocument_BeginPrint(object sender, PrintEventArgs e) {
            PrintDocument doc = sender as PrintDocument;
            string printerName = doc.PrinterSettings.PrinterName;
            // ... your code
        }

    }
}